### PR TITLE
Update Python docs

### LINKED ISSUE #157: Outdated/incomplete Python docs
Since @RoyStegeman started using it, we finally got another user that is not a developer (first one being @scarlehoff).
Nice chance to improve docs.

- [ ] `convolute` does not exist any longer (it is called `convolute_with_one`)
  https://github.com/N3PDF/pineappl/blob/552dff4780ce5e4c0e037474391e662bb92ef54c/pineappl_py/docs/source/recipes.rst#L16
- [ ] the call is also outdated: now the first argument now should be `pdg_id` (same call)

### PR DESCRIPTION
**Pull request by felixhekhorn**
Closes #157 

This mostly updates and fixes all Python doc strings. 

Apart from that there are few more (even breaking) changes:
- `PyGrid::evolve` got deprecated (as the underlying function)
- `PyFkTable::convolve_with_two`: return type was harmonized with `PyFkTable::convolve_with_one` to avoid the deprecation of `into_pyarray`
- the transition "Lumi" -> "Channel" was performed both on the PyO3 and Python side

if necessary we could also add other changes to the Python interface here, such as [to expose `Grid::convolutions`](https://github.com/NNPDF/pineko/pull/181#discussion_r1689352151)

### CODE REVIEW

**Review by Radonirinaunimi — COMMENTED** [review]
Thanks for this @felixhekhorn.

I will have a detailed look as soon as possible but I've picked up very tiny typos while I was skimming through.

Are you also  planning to modify the occurrences of `lumi(_)` in the argument of the functions and in the variables names of `src/fk_table.rs`, `src/grid.rs`, and `pineappl/grid.py`?

File: pineappl_py/pineappl/grid.py [hunk 1]
```diff
@@ -193,38 +191,38 @@ def convolve_with_one(
...
+            lhapdf like callable with arguments `pid, x, Q2` returning x*pdf for :math:`x`-grid
+        alphas : callable
+            lhapdf like callable with arguments `Q2` returning :math:`\alpha_s`
+        order_mask : sequence(bool)
+            Mask for selecting specific orders. The value `True` means the corresponding order
+            is included. An empty list corresponds to all orders being enabled.
+        bin_indices : sequence(int)
+            A list with the indices of the corresponding bins that should be calculated. An
+            empty list means that all orders should be calculated.
+        lumi_mask : sequence(bool)
```
Comment (cschwan) [inline pineappl_py/pineappl/grid.py:210]: Here you should write `channel_mask` or something similar. Also replace luminosity with channels in the text.

File: pineappl_py/pineappl/grid.py [hunk 2]
```diff
@@ -248,42 +246,42 @@ def convolve_with_two(
...
```
Comment (cschwan) [inline pineappl_py/pineappl/grid.py:269]: Same remarks as for `convolve_with_one` apply also here.

File: pineappl_py/pineappl/channel.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from .pineappl import PyChannel
```
Comment (cschwan) [inline pineappl_py/pineappl/channel.py:1]: We shouldn't explicitly create Python modules anymore and let PyO3 do that.

Comment (Radonirinaunimi) [reply pineappl_py/pineappl/channel.py:1]: Yes, indeed (cc https://github.com/NNPDF/pineappl/issues/200). I am planning to do this soon.

**Comment by cschwan** [comment]
#157 is a related Issue.

File: pineappl_py/pineappl/grid.py [hunk 4]
```diff
@@ -149,13 +147,13 @@ def __setitem__(self, key, subgrid):
 
         Parameters
         ----------
-            key : (int, int, int)
-                a 3-element integers tuple, consisting in `(order, bin, lumi)`
-            subgrid : ImportOnlySubgridV1
-                subgrid content
+        key : (int, int, int)
+            a 3-element integers tuple, consisting in `(order, bin, channel)`
```
[the remark this answers is not in the document]
Comment (felixhekhorn) [reply pineappl_py/pineappl/grid.py]: e3a08820

File: pineappl_py/pineappl/channel.py (hunk 3 above)
Comment (felixhekhorn) [reply pineappl_py/pineappl/channel.py:1]: addressed in #302

File: pineappl_py/pineappl/grid.py (hunk 1 above)
Comment (felixhekhorn) [reply pineappl_py/pineappl/grid.py]: 0d2866c2

File: pineappl_py/pineappl/grid.py (hunk 2 above)
Comment (felixhekhorn) [reply pineappl_py/pineappl/grid.py]: 0d2866c2

**Comment by felixhekhorn** [comment]
> Are you also planning to modify the occurrences of `lumi(_)` in the argument of the functions and in the variables names of `src/fk_table.rs`, `src/grid.rs`, and `pineappl/grid.py`?

I hope I caught all remaining occurrences - the only remaining appearance of the word "lumi" in `pineappl_py/src` is related to `pineappl::convolutions::LumiCache`, which I did not touch (as the underlying Rust is not touching it).

There is some overlap with #302 since both are touching the files in `pineappl_py/src`

**Comment by cschwan** [comment]
I ran `cargo fmt` in commit fd6f4226ecaf5dd57b253f2653c32844ca5e4b06. @felixhekhorn, could you check that these changes don't break anything?

**Comment by cschwan** [comment]
I can merge this now if you like, for wrappers of `Grid::{convolutions,set_convolutions}` we should open a new PR.

When this will be merged into `master`, I'm going to release this as a new minor version (0.9.0) instead of increasing the patch version of 0.8, because we renamed publicly-visible items.

**Comment by felixhekhorn** [comment]
> I ran `cargo fmt` in commit [fd6f422](https://github.com/NNPDF/pineappl/commit/fd6f4226ecaf5dd57b253f2653c32844ca5e4b06). @felixhekhorn, could you check that these changes don't break anything?

looks good

> we renamed publicly-visible items.

indeed, I know that the `channel_mask` parameter for `evolve` will break some of my scripts ...

**Comment by cschwan** [comment]
We need a strategy for merging this, since #302 removes all Python files and thus the changed documentation. So either I

1. merge this PR first, then #302, which means that in #302 the updated documentation has to be put in front of the Rust structs (probably),
2. or I merge #302 first, which means the changes from this PR will have to be rewritten here.

@felixhekhorn @Radonirinaunimi what path do you believe is best?

**Comment by Radonirinaunimi** [comment]
If this is ready it could be merged first and then I will resolve the conflicts in #302. The advantage of this is that this will address some of the todos in #302. If instead #302 is ready first then we reverse the order.

**Pull request merged**